packages/backendv2/src/middleware/access_control.ts

```diff
@@ -15,14 +17,30 @@ export const accessControl = (options?: AccessControlOptions) => {
     if (options?.unrestricted) {
       return next()
     }
+
+    const redisIsAvailable = redis !== null
+
+    const token: string =
+      ctx.headers.authorization.toLocaleLowerCase().replace("bearer ", "") || ""
+
+    // attempt retrieval of user from cache
     let user
-    try {
-      user = await getCurrentUserDetails(
-        ctx.headers.authorization.toLocaleLowerCase().replace("bearer ", ""),
-      )
-    } catch (error) {
-      throw new UnauthorizedError("unauthorized")
+    if (redisIsAvailable && redis.get) {
+      JSON.parse((await redis.get(token)) as string)
```

Comment:
Should we do something with this result?

Reply:
And what happens if redis.get returns undefined?

Reply:
> Should we do something with this result?

I removed the assignment by mistake. Fixed it.

> And what happens if redis.get returns undefined?

we check for it (and null) beneath - redis.get returns either a string or null